crates/configuration/src/version5/metadata/native_operations.rs

```diff
@@ -0,0 +1,376 @@
+//! Metadata information regarding native queries.
+
+// This code was copied from a different place that predated the introduction of clippy to the
+// project. Therefore we disregard certain clippy lints:
+#![allow(clippy::wrong_self_convention)]
+use super::database::*;
+
+use query_engine_sql::sql;
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+use std::fs;
+
+// Types
+
+/// Metadata information of Native Operations.
+#[derive(Debug, Clone, PartialEq, Eq, Default, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct NativeOperations {
+    /// Native Queries.
+    pub queries: NativeQueries,
+    /// Native Mutations.
+    pub mutations: NativeQueries,
+}
+
+/// Metadata information of Native Operations.
+#[derive(Debug, Clone, PartialEq, Eq, Default, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct NativeQueries(pub BTreeMap<String, NativeQueryInfo>);
+
+/// Information about a Native Operation
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct NativeQueryInfo {
+    /// SQL expression to use for the Native Operation.
+    /// We can interpolate values using `{{variable_name}}` syntax,
+    /// such as `SELECT * FROM authors WHERE name = {{author_name}}`
+    pub sql: NativeQuerySqlEither,
+    /// Columns returned by the Native Operation
+    pub columns: BTreeMap<String, ReadOnlyColumnInfo>,
+    #[serde(default)]
+    /// Names and types of arguments that can be passed to this Native Operation
+    pub arguments: BTreeMap<String, ReadOnlyColumnInfo>,
+    #[serde(default)]
+    pub description: Option<String>,
```

Comment:
There's no is procedure here anymore, since we split them to two different maps. This is generally only useful for ndc-spec schema generation to generate procedures or collections.